ndcube/meta.py

```diff
@@ -0,0 +1,374 @@
+
+        else:
+            new_meta = copy.deepcopy(self)
+            if isinstance(item, (numbers.Integral, slice)):
+                item = [item]
+            naxes = len(self.shape)
+            item = np.array(list(item) + [slice(None)] * (naxes - len(item)),
+                            dtype=object)
+
+            # Edit data shape and calculate which axis will be dropped.
+            dropped_axes = np.zeros(naxes, dtype=bool)
+            new_shape = new_meta.shape
+            for i, axis_item in enumerate(item):
+                if isinstance(axis_item, numbers.Integral):
+                    dropped_axes[i] = True
+                elif isinstance(axis_item, slice):
+                    start = axis_item.start
+                    if start is None:
+                        start = 0
+                    if start < 0:
+                        start = self.shape[i] - start
+                    stop = axis_item.stop
+                    if stop is None:
+                        stop = self.shape[i]
+                    if stop < 0:
+                        stop = self.shape[i] - stop
+                    new_shape[i] = stop - start
+                else:
+                    raise TypeError("Unrecognized slice type. "
+                                    "Must be an int, slice and tuple of the same.")
+            kept_axes = np.invert(dropped_axes)
+            new_meta._data_shape = new_shape[kept_axes]
+
+            # Slice all metadata associated with axes.
+            for key, value in self.items():
+                axis = self.axes.get(key, None)
+                drop_key = False
+                if axis is not None:
+                    # Calculate new axis indices.
+                    new_axis = np.asarray(list(
+                        set(axis).intersection(set(np.arange(naxes)[kept_axes]))
+                        ))
+                    if len(new_axis) == 0:
+                        new_axis = None
+                    else:
+                        cumul_dropped_axes = np.cumsum(dropped_axes)[new_axis]
+                        new_axis -= cumul_dropped_axes
+
+                    # Calculate sliced metadata values.
+                    axis_shape = tuple(self.shape[axis])
+                    if _is_scalar(value):
+                        new_value = value
+                        # If scalar metadata's axes have been dropped, mark metadata to be dropped.
+                        if new_axis is None:
+                            drop_key = True
+                    else:
+                        value_is_axis_aligned = _is_axis_aligned(value, axis_shape)
+                        if value_is_axis_aligned:
+                            new_item = kept_axes[axis]
+                        else:
+                            new_item = tuple(item[axis])
+                        # Slice metadata value.
+                        try:
+                            new_value = value[new_item]
+                        except:
```

Comment:
this needs an exception type. No Pokemon exceptions.

Reply:
Leave it up to the user to not apply a slice item that doesn't also work for the object they've entered as grid-aligned metadata.  So replace the `try...except...` block with L284.  (Confirm I understand the implications of this correcty.)

Reply:
Actually, this is needed for setting a tuple, list, etc. as axis-aligned metadata.  Otherwise, dropping multiple axes during slicing will not be supported.